1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,56 @@
 // Write your code here
+let slideIndex = 1;
+showSlides(slideIndex);
+let intervalId = null;
+
+function plusSlides(n) {
+  showSlides((slideIndex += n));
+}
+
+function currentSlide(n) {
+  showSlides((slideIndex = n));
+}
+
+function showSlides(n) {
+  let i;
+  let slides = document.getElementsByClassName("mySlides");
+  let dots = document.getElementsByClassName("dot");
+  if (n > slides.length) {
+    slideIndex = 1;
+  }
+  if (n < 1) {
+    slideIndex = slides.length;
+  }
+  for (i = 0; i < slides.length; i++) {
+    slides[i].style.display = "none";
+  }
+  for (i = 0; i < dots.length; i++) {
```

Comment:
I would just suggest using a conditional operator instead of a `for `loop could make the code more concise and potentially easier to read